crates/boundless-market/src/client.rs

```diff
@@ -99,33 +97,41 @@ impl<St, Si> ClientBuilder<St, Si> {
     /// Build the client
     pub async fn build(
         self,
-    ) -> Result<Client<StandardRpcProvider, St, StandardRequestBuilder<StandardRpcProvider, St>, Si>>
+    ) -> Result<Client<DynProvider, St, StandardRequestBuilder<DynProvider, St>, Si>>
```

Comment:
Note that these changes now allow the `Client` to be used without a private key. The Provider needs to be `dyn` so that it can be a different type depending on whether the private key is supplied. (It's also I think not a bad idea because it prevents changing this type from being a breaking change)